src/cpp/src/llm_pipeline.cpp

```diff
@@ -132,49 +186,32 @@ class LLMPipeline::LLMPipelineImpl {
         
         if (auto input_vector = std::get_if<std::vector<std::string>>(&inputs)) {
             encoded_input = m_tokenizer.encode(*input_vector);
-        } else if (auto input_str = std::get_if<std::string>(&inputs)) {
-            
-            std::string text = *input_str;
-            // todo: make for batched inputs as well
-            if (is_chat_conversation)
-                text = apply_chat_template(text);
-
-            // previous prompt generation in chat dialog stops with the end of sentence token, 
-            // need to append this token to the current prompt
-            if (is_chat_conversation && !m_is_cache_empty)
-                text = m_tokenizer.get_eos_token() + text;
+        } else if (auto input_prompt = std::get_if<std::string>(&inputs)) {
+            std::string& prompt = *input_prompt;
             
-            auto res = m_tokenizer.encode(text);
-            auto input_ids = res.input_ids;
-            auto attention_mask = res.attention_mask;
-
-            // todo: W/A If sentence begins with specfial tokens (<bos>, <s>, etc.) openvino_tokenizer inserts 2 special extra tokens <bos> and "▁",
-            // but HF does not do that. Moreover openvino_tokenizer always inserts <bos> but in chat scenario HF does not do that because skip_special_tokens=True.
-            // Need to remove both of that tokens manually to get exact token by token alignment with HF
-            auto size = input_ids.get_shape();
-            int64_t* inputs_data = input_ids.data<int64_t>();
-            std::vector<int64_t> tmp_ids(inputs_data, inputs_data + input_ids.get_size()); // todo: works only for batch 1
-
-            auto attention_mask_data = attention_mask.data<int64_t>();
-            std::vector<float> tmp_attn_mask(attention_mask_data, attention_mask_data + attention_mask.get_size());
-
-            std::vector<std::string> prefixes_to_exclude = {m_tokenizer.get_eos_token(), m_tokenizer.get_bos_token()};
-            auto prefix_match = [&text](std::string prefix) { return text.substr(0, prefix.length()) == prefix; };
-            if (std::any_of(prefixes_to_exclude.begin(), prefixes_to_exclude.end(), prefix_match)) {
-                tmp_ids.erase(tmp_ids.begin());
-                tmp_attn_mask.erase(tmp_attn_mask.begin());
+            if (is_chat_conversation) {
+                m_history.push_back({{"role", "user"}, {"content", prompt}});
```

Comment:
how user can pass its own input, which has been applied with chat template ?

Reply:
I think the initial idea was to require always calling `apply_chat_template()` before passing to `generate()`, right? Just in case, we could have yet another overload for `generate()` to detect if the template was applied by accepting a wrapper over a string returned from `apply_chat_template()`.

In any case I'd propose to ignore that for now.